assets/js/steps/utils/barcode-scanner.js

```diff
@@ -3,9 +3,22 @@
  * Contains the barcode scanning listener.
  * Based on the method from https://stackoverflow.com/a/55251571
  */
+import {
+    BARCODE_CODE_2BY5,
+    BARCODE_CODE_COMMAND,
+    BARCODE_TYPE_2BY5,
+    BARCODE_TYPE_COMMAND,
+    BARCODE_TYPE_DEFAULT
+} from '../../constants';
 
 // Barcode pattern.
-const pattern = /^(!BA11|!BA10|!BA)(?<code>.+)!C$/;
+const pattern = /^(!B[A-Z]\d{2})(?<code>.+)!C$/;
```

Comment:
If we match everything here is there an unknown barcode type handling?

Reply:
The plan was to leave that up to the engine to handle.
This is mostly to get the code from inside the barcode if the scanner is coded.